### PR TITLE
fetch: resp body shouldn't buffer if content-type is missing

### PR DESCRIPTION
**Pull request by lucacasonato**
Adds a test for https://bugzilla.mozilla.org/show_bug.cgi?id=1544313

Works fine in Chromium & WebKit.

cc @annevk﻿

### CODE REVIEW

**Comment by lucacasonato** [comment]
I assume the `infrastructure/ tests` failure is a flake. This PR does not touch any infra related code.

File: fetch/api/basic/stream-response.any.js [hunk 1]
```diff
@@ -26,4 +26,17 @@ promise_test(function(test) {
...
+  return fetch(RESOURCES_DIR + "trickle.py?ms=300&count=10&notype=true").then(function(resp) {
+    var count = 0;
+    if (resp.body)
+      return streamBody(resp.body.getReader(), test, count);
+    else
+      test.step(function() {
+        assert_unreached( "Body does not exist in response");
+      });
+  });
+}, "Stream response's body when content-type is not present");
```
Comment (annevk) [inline fetch/api/basic/stream-response.any.js:42]: Add newline.

File: fetch/api/basic/stream-response.any.js [hunk 2]
```diff
@@ -26,4 +26,17 @@ promise_test(function(test) {
         assert_unreached( "Body does not exist in response");
       });
   });
-}, "Stream response's body");
+}, "Stream response's body when content-type is present");
+
+// This test makes sure that the response body is not buffered if no content type is provided.
+promise_test(function(test) {
+  return fetch(RESOURCES_DIR + "trickle.py?ms=300&count=10&notype=true").then(function(resp) {
+    var count = 0;
```
Comment (annevk) [inline fetch/api/basic/stream-response.any.js:34]: Both here and above maybe drop this and just pass 0 to streamBody or make that argument optional even?

File: fetch/api/basic/stream-response.any.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 // META: global=window,worker
 // META: script=../resources/utils.js
 
-function streamBody(reader, test, count) {
+function streamBody(reader, test) {
+  let count = 0;
```
Comment (annevk) [inline fetch/api/basic/stream-response.any.js:5]: I think you need to keep this as a third argument with a default value, given the recursive invocation. Or am I missing something?

Comment (lucacasonato) [reply fetch/api/basic/stream-response.any.js:5]: Oh right... not properly awake yet it seems.

**Review by annevk — APPROVED** [review]

**Pull request merged**